src/menu/src/Menu.js

```diff
@@ -7,112 +7,114 @@ import MenuGroup from './MenuGroup'
+              return menuItems.current[indexOfItem + 1]
+            }
 
-      // Helper function for getting next legitimate element
-      const move = elem => {
-        const indexOfItem = this.menuItems.indexOf(elem)
+            return startItem
+          }
 
-        if (goingDown) {
-          if (indexOfItem < this.menuItems.length - 1) {
-            return this.menuItems[indexOfItem + 1]
+          if (indexOfItem - 1 > -1) {
+            return menuItems.current[indexOfItem - 1]
           }
 
           return startItem
         }
 
-        if (indexOfItem - 1 > -1) {
-          return this.menuItems[indexOfItem - 1]
-        }
+        // Make first move
+        let nextItem = move(currentItem)
 
-        return startItem
-      }
-
-      // Make first move
-      let nextItem = move(currentItem)
+        // If the menuitem is disabled move on
+        while (nextItem.disabled) {
+          nextItem = move(nextItem)
+        }
 
-      // If the menuitem is disabled move on
-      while (nextItem.disabled) {
-        nextItem = move(nextItem)
+        // Focus the first one that's not disabled
+        nextItem.focus()
       }
 
-      // Focus the first one that's not disabled
-      nextItem.focus()
-    }
-
-    this.menuItems.forEach(menuItem => {
-      // Handle key presses for menuItem
-      menuItem.addEventListener('keydown', e => {
-        // Go to next/previous item if it exists
-        // or loop around
+      menuItems.current.forEach(menuItem => {
+        // Handle key presses for menuItem
+        menuItem.addEventListener('keydown', e => {
+          // Go to next/previous item if it exists
+          // or loop around
 
-        if (e.key === 'ArrowDown') {
-          e.preventDefault()
-          focusNext(menuItem, this.firstItem)
-        }
+          if (e.key === 'ArrowDown') {
+            e.preventDefault()
+            focusNext(menuItem, firstItem.current)
+          }
 
-        if (e.key === 'ArrowUp') {
-          e.preventDefault()
-          focusNext(menuItem, this.lastItem)
-        }
+          if (e.key === 'ArrowUp') {
+            e.preventDefault()
+            focusNext(menuItem, lastItem.current)
+          }
 
-        if (e.key === 'Home') {
-          e.preventDefault()
-          this.firstItem.focus()
-        }
+          if (e.key === 'Home') {
+            e.preventDefault()
+            firstItem.current.focus()
+          }
 
-        if (e.key === 'End') {
-          e.preventDefault()
-          this.lastItem.focus()
-        }
+          if (e.key === 'End') {
+            e.preventDefault()
+            lastItem.current.focus()
+          }
+        })
       })
-    })
-  }
-
-  onMenuRef = ref => {
-    this.menuRef = ref
-  }
-
-  render() {
-    const { children } = this.props
-    return (
-      <Pane is="nav" innerRef={this.onMenuRef} role="menu" outline="none">
-        {children}
-      </Pane>
-    )
-  }
+    }
+  }, [menuRef])
```

Comment:
This will only trigger a re-run when the actual ref object changes, not when `menuRef.current` changes. Is that intended?